### PR TITLE
Do not use extra module for pkg/eraser

### PR DESCRIPTION
**Pull request by cpuguy83**
**What this PR does / why we need it**:

    Multi-module repos are highly problematic for go currently and
    complicate development.
    Tools like gopls do not support it breaks the normal dev workflow.
    
    Instead just keep everything in one module for now.
    We may return to multi-module once the repo is stabilized or go is in a
    better place for multi-module development.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,29 +1,47 @@
...
-# Build
+FROM builder AS eraser-build
 RUN \
-    --mount=type=cache,target=/root/gocache \
-    GOCACHE=/root/gocache CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
+    --mount=type=cache,target=${GOCACHE} \
+    --mount=type=cache,target=/go/pkg/mod \
+    GOOS=linux GOARCH=amd64 go build -ldflags '-w -extldflags "-static"' -o out/removal ./pkg/eraser
+
+FROM scratch as eraser
```
Comment (sozercan) [inline Dockerfile]: can we just use distroless/static:nonroot here? so we don't have to worry about updating/copying certs

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
+COPY --from=eraser-build /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
+COPY --from=eraser-build /workspace/out/removal /
+ENTRYPOINT ["/removal"]
```
Comment (sozercan) [inline Dockerfile]: should we rename this to eraser?

File: pkg/eraser/Dockerfile [hunk 3]
```diff
@@ -1,22 +1,23 @@
-FROM golang:1.17 AS build
+# syntax=mcr.microsoft.com/oss/moby/dockerfile:1.3.1
```
Comment (sozercan) [inline pkg/eraser/Dockerfile]: do we still need this file?

Comment (cpuguy83) [reply pkg/eraser/Dockerfile]: I could have sworn I deleted this one, guess I missed it

File: Dockerfile [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
 WORKDIR /
-COPY --from=builder /workspace/manager .
+COPY --from=manager-build /workspace/out/manager .
 USER 65532:65532
-
 ENTRYPOINT ["/manager"]
+
+FROM scratch AS bins
+COPY --from=manager-build /workspace/out/manager /
+COPY --from=eraser /workspace/out/removal /
```
Comment (sozercan) [inline Dockerfile]: nit: newline

**Comment by sozercan** [comment]
not sure why cla is not green for you

File: Dockerfile (hunk 1 above)
Comment (cpuguy83) [reply Dockerfile]: Fixed

File: pkg/eraser/Dockerfile (hunk 3 above)
Comment (cpuguy83) [reply pkg/eraser/Dockerfile]: Fixed

File: Dockerfile (hunk 4 above)
Comment (cpuguy83) [reply Dockerfile]: Fixed

File: Dockerfile [hunk 5]
```diff
@@ -1,29 +1,45 @@
...
-FROM gcr.io/distroless/static:nonroot
+FROM gcr.io/distroless/static:nonroot AS manager
 WORKDIR /
-COPY --from=builder /workspace/manager .
+COPY --from=manager-build /workspace/out/manager .
 USER 65532:65532
-
 ENTRYPOINT ["/manager"]
+
+FROM scratch AS bins
```
Comment (sozercan) [inline Dockerfile]: what is this used for?

**Review by sozercan — APPROVED** [review]
comment above, otherwise LGTM. Thanks!

**Pull request merged**